src/service.rs

```diff
@@ -1368,13 +1368,30 @@ impl Service {
 
     /// The equivalent of libp2p `inject_connected()` for a udp session. We have no stream, but a
     /// session key-pair has been negotiated.
-    fn inject_session_established(&mut self, enr: Enr, direction: ConnectionDirection) {
+    fn inject_session_established(
+        &mut self,
+        enr: Enr,
+        direction_instruction: ConnectionDirectionInstruction,
+    ) {
         // Ignore sessions with non-contactable ENRs
         if self.ip_mode.get_contactable_addr(&enr).is_none() {
             return;
         }
 
         let node_id = enr.node_id();
+
+        let direction = match direction_instruction {
+            ConnectionDirectionInstruction::Incoming => ConnectionDirection::Incoming,
+            ConnectionDirectionInstruction::Outgoing => ConnectionDirection::Outgoing,
+            ConnectionDirectionInstruction::IncomingIfNotExists => {
+                let key = kbucket::Key::from(node_id);
+                match self.kbuckets.write().entry(&key) {
+                    Entry::Present(_, status) if status.is_connected() => status.direction,
+                    _ => ConnectionDirection::Incoming,
```

Comment:
we also have the `NodeStatus` in a `Pending` entry, should we include it here? Not a suggestion, but a question